### PR TITLE
C++/ROS Client Documentation

### LINKED ISSUE #99: Create documentation for C++ and ROS clients
Create markdown documentation for C++ and ROS clients so it can be put up onto the documentation site quickly.

### PR DESCRIPTION
**Pull request by bradleynull**
This is the documentation for the C++ and ROS clients per their current state. I essentially translated each README, correct errors, and added more exposition for the quick starts. Each example page is broken into two sections

* Description of generic means of connecting and using the simulator and sensors
* Specific discussion of each individual example

Closes #99

### CODE REVIEW

File: docs/cpp_client/cpp_examples.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+## Replay Example
+
+A full example of controlling the EGO vehicle in Replay mode can be found in 
+`monodrive-client/cpp-client/cpp-examples/replay/replay_example.cpp`. This 
+example demonstrates:
+
+* Configuring and connecting to a running instance of the monoDrive Simulator
+* Configuring and connecting to a Camera sensor (as discussed above)
+* Configuring the Viewport Sensor for the camera on the simulator
+* Stepping the a monoDrive Trajectory File in Replay mode (as discussed above)
```
Comment (katiemacaulay) [inline docs/cpp_client/cpp_examples.md:117]: Get rid of the or a like:  Stepping a monoDrive Trajectory File in Replay mode (as discussed above)

File: docs/cpp_client/cpp_examples.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+The Viewport Sensor in this example controls where the camera view will be 
+placed in the monoDrive Simulator or Scenario Editor. This sensor is not 
+sampled, but rather just configured and sent to the `sim0` instance:
```
Comment (katiemacaulay) [inline docs/cpp_client/cpp_examples.md:121]: Take out just: rather configured and sent to the `sim0` instance:

File: docs/cpp_client/cpp_examples.md (hunk 1 above)
Comment (bradleynull) [reply docs/cpp_client/cpp_examples.md:117]: Fixed

File: docs/cpp_client/cpp_examples.md (hunk 2 above)
Comment (bradleynull) [reply docs/cpp_client/cpp_examples.md:121]: Changed to: "This sensor is not configured and sent to the `sim0` instance without needing to be sampled"

**Comment by bradleynull** [comment]
@katiemacaulay Thanks for the comments! Ready for another look!

**Pull request merged**